consensus/spos/consensusState.go

```diff
@@ -24,6 +25,9 @@ type ConsensusState struct {
 	receivedHeaders    []data.HeaderHandler
 	mutReceivedHeaders sync.RWMutex
 
+	receivedMessagesWithSignature    map[string]p2p.MessageP2P
```

Comment:
...InvalidSigners ?

Reply:
it will actually store all the messages with signature, and later on when we check the invalid signers we get the invalid ones from this map

Reply:
ok